src/TicTacToe.ConsoleApp/GameEngine.cs

```diff
@@ -0,0 +1,144 @@
+                    {
+                        Console.WriteLine("Game over! \n");
+                        Console.WriteLine($"{game.OpponentUsername} Won!");
+                        Console.WriteLine($"{game.CreatorUsername} Lost!");
+                    }
+                    else if (game.State == GameState.Draw)
+                    {
+                        Console.WriteLine("Game over!");
+                        Console.WriteLine("It's Draw");
+                    }
+                    else if ((game.State == GameState.CreatorTurn && game.CreatorUserId == userId) || (game.State == GameState.OpponentTurn && game.OpponentUserId == userId))
+                    {
+                        Console.WriteLine("It's my turn...");
+
+                        while (int.TryParse(Console.ReadLine(), out var input))
+                        {
+                            var choosedPosition = ChoosePosition(input);
+                            gameService.Play(gameId, userId, choosedPosition.Row, choosedPosition.Col);
+                            break;
+                        }
+                    }
+                    else
+                    {
+                        Console.WriteLine("It's opponent turn...");
+                    }
+                }
+
+                Thread.Sleep(250);
+                Console.Clear();
+            }
+        }
+
+        public Guid GetOrCreateGame(Guid userId)
+        {
+            using (var context = new TicTacToeDbContextFactory().CreateDbContext())
+            {
+                var gameService = new GameService(context, new GameResultValidator());
+                var availableGames = gameService.GetAvailableGames(userId).ToList();
+
+                if (!availableGames.Any())
+                {
+                    Console.WriteLine("There are no available games, please create one");
+                    Console.Write("Enter name of the game: ");
+
+                    var gameName = Console.ReadLine();
+                    var newGame = new GameCreationInput() { Name = gameName, Visibility = VisibilityType.Public };
+
+                    return gameService.Create(newGame, userId).Id;
+                }
+                else
+                {
+                    Console.WriteLine("Choose game:");
+
+                    for (int i = 0; i < availableGames.Count; i++)
+                    {
+                        Console.WriteLine($"{i+1}. {availableGames[i].Name}");
+                    }
+
+                    int chosenGame;
+                    while (!int.TryParse(Console.ReadLine(), out chosenGame) || chosenGame < 1 || chosenGame > availableGames.Count)
+                    {
+                        Console.WriteLine("Enter valid number");
+                    }
+
+                    return gameService.Join(availableGames[chosenGame - 1].Id, userId).Id;
+                }
+            }
+        }
+
+        private void PrintBoard(string board)
+        {
```

Comment:
Optimize the method `PrintBoard()` to use `StringBuilder`.

Reply:
done